### ISSUE TITLE
Build Generated POM files still has dependencies like Apache HTTP client and Guava

### ISSUE THREAD

**Issue by dhaval24** [Microsoft/ApplicationInsights-Java#583]
The POMs generated via build has multiple dependencies inside which are shaded. 

```
<dependencies>
    <dependency>
      <groupId>eu.infomas</groupId>
      <artifactId>annotation-detector</artifactId>
      <version>3.0.5</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>commons-io</groupId>
      <artifactId>commons-io</artifactId>
      <version>2.6</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.commons</groupId>
      <artifactId>commons-lang3</artifactId>
      <version>3.7</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.httpcomponents</groupId>
      <artifactId>**httpclient**</artifactId>
      <version>4.5.3</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>**guava**</artifactId>
      <version>20.0</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>com.google.code.gson</groupId>
      <artifactId>gson</artifactId>
      <version>2.8.2</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>4.12</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.mockito</groupId>
      <artifactId>mockito-all</artifactId>
      <version>1.10.19</version>
      <scope>test</scope>
    </dependency>
  </dependencies>
```

Excerpt from pom file for Core.

Especially Guava and HTTP Client are problematic and can lead to class path hell issues. We should fix the POM script to remove this.

**Comment by dhaval24** [comment]
https://github.com/micrometer-metrics/micrometer/issues/441. See the comments

**Comment by littleaj** [comment]
Fixed in #582
